stainless_backend/tests/examples/mod.rs

```diff
@@ -36,6 +36,7 @@ pub fn identity_symbols<'a>(f: &'a Factory) -> Symbols<'a> {
   functions.insert(fd.id, fd);
   Symbols {
     sorts: Map::new(),
```

Comment:
Are we actually extracting any ADTs/sorts or will `sorts` always be empty? In the Scala front-end, we only ever extract classes and those eventually get lowered to ADTs.

If so, perhaps we can just remove this field altogether?

Reply:
For the moment, ADTs _are_ extracted as sorts in the Rust-frontend. 
https://github.com/epfl-lara/rust-stainless/blob/master/stainless_extraction/src/krate.rs#L559:
```rust
f.ADTSort(adt_id, tparams, constructors, flags)
```

They are then transformed to classes [in the `NoxtFrontend`](https://github.com/epfl-lara/stainless/blob/rust-interop/frontends/stainless-noxt/src/main/scala/stainless/frontends/noxt/NoxtFrontend.scala#L108) on the Scala-Stainless side. We'd have to ask @gsps for the reason for this.


However, with this PR we introduce _actual class extraction_, so we might shift the transformation from sorts to classes into the frontend and only emit classes?

Reply:
That makes sense, since Rust structs and enums are actual ADTs unlike the class-based Scala 2 encoding. I don't see the need to extract them as classes as this would only introduce overhead in the various `oo` phases for no actual gains that I can see. Let's leave things like that then :)